shares/parse_sparse_shares.go

```diff
@@ -56,27 +59,29 @@ func parseSparseShares(shares []Share, supportedShareVersions []uint8) (blobs []
 			if err != nil {
 				return nil, err
 			}
-			blob := blob.New(ns, data, version)
 			sequences = append(sequences, sequence{
-				blob:        blob,
-				sequenceLen: sequenceLen,
+				ns:           ns,
+				shareVersion: version,
+				data:         data,
+				sequenceLen:  sequenceLen,
 			})
 		} else { // continuation share
 			if len(sequences) == 0 {
 				return nil, fmt.Errorf("continuation share %v without a sequence start share", share)
 			}
+			// FIXME: it doesn't look like we check whether all the shares belong to the same namespace.
```

Comment:
This PR ?

Reply:
Are you implying I should fix it in this PR?

Reply:
Nah, just asking. Thanks for filling the issues https://github.com/celestiaorg/go-square/issues/78